app/web/src/components/ModelingDiagram/diagram_types.ts

```diff
@@ -408,6 +411,9 @@ export class DiagramEdgeData extends DiagramElementData {
   }
 
   get toSocketKey() {
+    if (!this.def.toSocketId) {
```

Comment:
Oh, we probably don't need these exceptions anymore now that DiagramSocketEdgeData is a subclass with socket-speific `def`--we should get rid of them to avoid confusion.